src/services/sync-component-with-file/config-file-changes-transformer.ts

```diff
@@ -0,0 +1,110 @@
+import { ComponentChanges, ComponentSyncPayload, ComponentUnlinkPayload, ModifiedFilePayload } from '../../types';
+import { listFeatures } from '../feature-flags';
+
+const isApplicableForBothIdentifiersOrOnlyForId = (isApplicableForId: boolean) => {
+  const { isCreateFromYamlEnabled } = listFeatures();
+
+  return isCreateFromYamlEnabled || isApplicableForId;
+};
+
+const isFileMovedDetectionEnabledForName = (isApplicableForName: boolean) => {
+  const { isCreateFromYamlEnabled } = listFeatures();
+
+  return isCreateFromYamlEnabled && isApplicableForName;
+};
+
+const isFileIdentifierChanged = (oldFile: ComponentUnlinkPayload, newFile: ComponentSyncPayload): boolean => {
+  const { isCreateFromYamlEnabled } = listFeatures();
+  const { id: oldId, name: oldName } = oldFile.componentYaml;
+  const { id: newId, name: newName } = newFile.componentYaml;
+
+  const isIdChanged = oldId !== newId;
+  const isNameFileIdentifier = oldId === undefined && newId === undefined;
+  const isIdAdded = oldId === undefined && newId !== undefined;
+
+  const isFileMovedAsGitlabRename = newFile.filePath !== newFile.previousFilePath;
+  const isNameChangedWithFileMove = isNameFileIdentifier && isFileMovedAsGitlabRename && oldName !== newName;
+  const isIdFileIdentifierChanged = isIdChanged && !isIdAdded;
+
+  return isCreateFromYamlEnabled ? isNameChangedWithFileMove || isIdFileIdentifierChanged : isIdChanged;
+};
+
+export const detectMovedFilesAndUpdateComponentChanges = (changes: ComponentChanges): ComponentChanges => {
+  let creates = [...changes.componentsToCreate];
+  const updates = [...changes.componentsToUpdate];
+  let unlinks = [...changes.componentsToUnlink];
+
+  creates.forEach(({ componentYaml, absoluteFilePath, filePath }) => {
+    const { id: createId, name: createName } = componentYaml;
+
+    let isMovedById = false;
+    let isMovedByName = false;
+    let isMovedByIdentifierTransition = false;
+
+    const movedFile = unlinks.find(({ componentYaml: { id: removeId, name: removeName } }) => {
+      const isCreateId = Boolean(createId);
```

Comment:
Do we have test to capture all use-cases for the 4 variables leading to 3 isMovedBy results? looking at the test we don't have covered all possible permutations here.

Reply:
I think [find-config-file-changes.test.ts](https://github.com/atlassian-labs/gitlab-for-compass/pull/34/files#diff-d4538fafa37df81717c6eb8697b57cbfd574c5cae987f9a64958fe0087b216f5) had a lot of test cases that covered the important cases that came to mind for me. Which permutations should we add to the test cases?